src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Persistence/Services/DelegationService.cs

```diff
@@ -0,0 +1,421 @@
+    private readonly IAssignmentPackageRepository assignmentPackageRepository = assignmentPackageRepository;
+    private readonly IAssignmentResourceRepository assignmentResourceRepository = assignmentResourceRepository;
+    private readonly IRolePackageRepository rolePackageRepository = rolePackageRepository;
+    private readonly IRoleResourceRepository roleResourceRepository = roleResourceRepository;
+    private readonly IDelegationRepository delegationRepository = delegationRepository;
+    private readonly IPackageRepository packageRepository = packageRepository;
+    private readonly IResourceRepository resourceRepository = resourceRepository;
+    private readonly IDelegationPackageRepository delegationPackageRepository = delegationPackageRepository;
+    private readonly IDelegationResourceRepository delegationResourceRepository = delegationResourceRepository;
+    private readonly IAssignmentService assignmentService = assignmentService;
+    private readonly IEntityRepository entityRepository = entityRepository;
+    private readonly IConnectionRepository connectionRepository = connectionRepository;
+    private readonly IEntityTypeRepository entityTypeRepository = entityTypeRepository;
+    private readonly IEntityVariantRepository entityVariantRepository = entityVariantRepository;
+    private readonly IEntityLookupRepository entityLookupRepository = entityLookupRepository;
+    private readonly IConnectionPackageRepository connectionPackageRepository = connectionPackageRepository;
+
+    private async Task<bool> CheckIfEntityHasRole(string roleCode, Guid fromId, Guid toId)
+    {
+        var role = (await roleRepository.Get(t => t.Code, roleCode)).First();
+
+        var filter = inheritedAssignmentRepository.CreateFilterBuilder();
+        filter.Equal(t => t.FromId, fromId);
+        filter.Equal(t => t.ToId, toId);
+        filter.Equal(t => t.RoleId, role.Id);
+
+        var userAssignments = await assignmentRepository.Get();
+
+        if (userAssignments == null || !userAssignments.Any())
+        {
+            return false;
+        }
+        return true;
+    }
+
+    /// <inheritdoc/>
+    public async Task<ExtDelegation> CreateDelgation(Guid userId, Guid fromAssignmentId, Guid toAssignmentId)
+    {
+        var fromAssignment = await assignmentRepository.GetExtended(fromAssignmentId);
+        var toAssignment = await assignmentRepository.GetExtended(toAssignmentId);
+
+        // Sjekk om from og to deler en felles entitet
+        if (fromAssignment.ToId != toAssignment.FromId) 
+        {
+            throw new InvalidOperationException("Assignments are not connected. FromAssignment.ToId != ToAssignment.FromId");
+        }
+
+        // Sjekk om bruker er Tilgangsstyrer for FromAssignment
+        // TODO: Sjekk inheireted. Man kan få TS fra DAGL
+        var assResTS = await assignmentService.GetAssignment(fromAssignment.ToId, userId, "TS");
```

Comment:
Samme problem her i tillegg til at TS er feil rolle for klientdelegering.

Delegation krever da Klientadministrasjon, som vi har laget systemressurs for som sjekkes på kontroller nivå, hvor man får tilgang via både A2 Klientadministrator rolle og ny A3 Klientadministrator rolle.